docs/ingester-handover.md

```diff
@@ -0,0 +1,44 @@
+# Ingester Hand-over
+
+The [ingester](architecture.md#ingester) holds several hours of sample
+data in memory. When we want to shut down an ingester, either for
+software version update or to drain a node for maintenance, this data
+must not be discarded.
+
+Each ingester goes through different states in its lifecycle. When
+working normally, the state is `ACTIVE`.
+
+On start-up, an ingester first goes into state `PENDING`. After a
+short time, if nothing happens, it adds itself to the ring and goes
+into state ACTIVE.
+
+A running ingester is notified to shut down by Unix signal
+`SIGINT`. On receipt of this signal it goes into state `LEAVING` and
+looks for an ingester in state `PENDING`. If it finds one, that
+ingester goes into state `JOINING` and the leaver transfers all its
+in-memory data over to the joiner. On successful transfer the leaver
+removes itself from the ring and exits and the joiner changes to
+`ACTIVE`, taking over ownership of the leaver's
+[ring tokens](architecture.md#hashing).
+
+If a leaving ingester does not find a pending ingester, it will flush
```

Comment:
May want to mention this timeout is tuneable with a flag (`-ingester.join-after`

Reply:
Ah no, its `ingester.max-transfer-retries` - it will try, backoff for 5 secs (max).

Reply:
Yeah if you win the lottery.  See #1307